burnman/mineral.py

```diff
@@ -64,6 +65,15 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
+            # Reset everything but params
+            for key, elem in self.__dict__.items():
+                if key != 'params':
+                    elem = float('nan')
+
```

Comment:
I still think that the way this is written is way too dangerous.  For instance, will this set self.method to NaN?  What if somebody subclasses mineral and adds a few new members that are not P,T dependent?

As an alternative, take a look at what Timo has done in the performance improvements branch (i.e., manually setting the derived properties to None).  This is more explicit and less likely to break things in unexpected ways.